### PR TITLE
stuff for clang/LLVM

### PR DESCRIPTION
**Pull request by fido2020**

### CODE REVIEW

**Comment by avdgrinten** [comment]
#214 added wait4 concurrently. Are the off64_t functions exported by other libcs (like musl or glibc)?

**Comment by fido2020** [comment]
I noticed that the wait4 added just throws an assertion, would it not be better to just call waitpid? It would be better I replace lseek64 with a define as that appears to be the way musl defines it. glibc's lseek64 is system dependent.

**Comment by avdgrinten** [comment]
I would actually prefer proper type defs and a proper lseek64 function over macros. However, it is not clear to me that they should be defined at all, since they seem to be compatibility hacks on 32-bit systems. Does llvm call them directly?

File: sysdeps/lemon/generic/lemon.cpp [hunk 1]
```diff
@@ -72,6 +72,8 @@ namespace mlibc{
 
 	void sys_exit(int status){
 		syscall(SYS_EXIT, status);
+
+		for(;;); // Prevent compiler warning
```
Comment (Geertiebear) [inline sysdeps/lemon/generic/lemon.cpp:76]: This could also be done with ``__builtin_unreachable`` no?

Comment (fido2020) [reply sysdeps/lemon/generic/lemon.cpp]: Yes, fixeed

**Comment by fido2020** [comment]
LLVM calls it once, I could just patch my LLVM if you would prefer that?

**Review by avdgrinten — APPROVED** [review]
LGTM.

**Pull request merged**